pyam/utils.py

```diff
@@ -208,8 +208,29 @@ def style_df(df, style='heatmap'):
         return df.style.background_gradient(cmap=cm)
 
 
-def find_depth(data, s, level):
-    # determine function for finding depth level =, >=, <= |s
+def find_depth(data, s='', level=None):
+    """
+    return a list of bools asserting the depth indicated by `level`;
+    or if level is None, return -1 if the string does not contain `s` and
+    the depth (number of `|`) otherwise.
+    string `s` is removed from `data` if at the start of a variable.
+    """
+    # remove wildcard as last character from string, find depth
+    s = s.rstrip('*')
+    pipe = re.compile('\\|')
+    regexp = re.compile(_escape_regexp(s))
+
+    def _find_depth(val):
+        return len(pipe.findall(re.sub(regexp, '', val)))
+
+    # if no level test is specified, return the depth as int
+    if level is None:
+        def return_depth(val):
+            return -1 if not val.startswith(s) else _find_depth(val)
```

Comment:
Would highly suggest returning either `None` or `np.nan` or `False` instead of `-1`.

Reply:
refactor to `None`